apps/prairielearn/src/lib/socket-server.ts

```diff
@@ -95,6 +95,6 @@ export async function close() {
   io.engine.close();
 
   // Shut down the Redis clients.
-  await pub?.quit();
-  await sub?.quit();
+  pub?.disconnect();
+  sub?.disconnect();
```

Comment:
These and similar changes are drive-by fixes to some Redis shutdown issues I saw. Per https://redis.io/docs/latest/commands/quit/, `quit` is deprecated and one should just disconnect instead.

Reply:
Actually, these aren't quite interchangeable. Docs for `disconnect()`:

> This method closes the connection immediately, and may lose some pending replies that haven't written to client. If you want to wait for the pending replies, use Redis#quit instead.

Reply:
And somehow, these changes seem to cause tests to break :( it seems to be a timing issue somehow. The async call to `quit()` seems to take just long enough for the timing issue to not manifest. I'll back this change out.

Reply:
Undone in c3eb187.